backends/s3/credentials.go

```diff
@@ -0,0 +1,53 @@
+package s3
+
+import (
+	"os"
+
+	"github.com/minio/minio-go/v7/pkg/credentials"
+)
+
+// K8sSecretProvider is an implementation of Minio's credentials.Provider,
+// allowing to read credentials from Kubernetes secrets, as described in
+// https://kubernetes.io/docs/tasks/inject-data-application/distribute-credentials-secure.
+type K8sSecretProvider struct {
+	// Path to the fiel containing the access key,
+	// e.g. /etc/s3-secrets/access-key.
+	AccessKeyFilename string `json:"access_key_file"`
+
+	// Path to the fiel containing the secret key,
+	// e.g. /etc/s3-secrets/secret-key.
+	SecretKeyFilename string `json:"secret_key_file"`
+}
+
+// IsExpired implements credentials.Provider.
+// As there is no totally reliable way to tell
+// if a file was modified accross all filesystems except opening it,
+// we always return true, and p.Retrieve will open it regardless.
+func (*K8sSecretProvider) IsExpired() bool {
+	return true
```

Comment:
This will tell the minio client to read the files every time. Opening and reading two files for every single request can quickly become a bottleneck.

Instead, please cache the contents for a second. A second will not cause any practical issues and the performance gain of caching something that can happen hundreds of times per second for a second far outweighs the gain of further increasing it to say 30 seconds.

Reply:
There is the added issue that the minio client may decide to reconnect every time you return `true` here, which is every single time in this case. To avoid this, when the cache time has expired, load the files from disk and check if they have actually changed before returning true here.